src/main/perf/SearchPerfTest.java

```diff
@@ -275,6 +279,56 @@ private static void _main(String[] clArgs) throws Exception {
       throw new UnsupportedOperationException("recacheFilterDeletes was deprecated");
     }
 
+    FacetsConfig facetsConfig = new FacetsConfig();
+    facetsConfig.setHierarchical("Date.taxonomy", true);
+    facetsConfig.setHierarchical("Date.sortedset", true);
+
+    // all unique facet group fields ($facet alone, by default):
+    final Set<String> facetFields = new HashSet<>();
+
+    // facet dim name -> facet method
```

Comment:
Could you also fix `SearchPerfTest` to pre-compute the necessary reader states before kicking off the searcher threads for actual benchmarking?  `SearchPerfTest` is the entry point for all the nightly performance tasks.

Reply:
The newest commit does this and is a more useful example of what we are trying to illustrate.